Extensions/ReadPixels.json

```diff
@@ -0,0 +1,238 @@
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": true
+        }
+      ],
+      "parameters": [
+        {
+          "codeOnly": false,
+          "defaultValue": "",
+          "description": "Position X",
+          "longDescription": "",
+          "name": "PositionX",
+          "optional": false,
+          "supplementaryInformation": "",
+          "type": "expression"
+        },
+        {
+          "codeOnly": false,
+          "defaultValue": "",
+          "description": "Position Y",
+          "longDescription": "",
+          "name": "PositionY",
+          "optional": false,
+          "supplementaryInformation": "",
+          "type": "expression"
+        }
+      ],
+      "objectGroups": []
+    },
+    {
+      "description": "",
+      "fullName": "",
+      "functionType": "Action",
+      "name": "onFirstSceneLoaded",
+      "sentence": "",
+      "events": [
+        {
+          "disabled": false,
+          "folded": false,
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "\n// Define an new private object javascript for this extension\ngdjs._readPixels = {};\n\n// getCanvasType, the argument is HTMLCanvasElement\ngdjs._readPixels.getCanvasType = function (canvas) {\n    const is2d = canvas.getContext(\"2d\");\n    const isBitmap = canvas.getContext(\"bitmaprenderer\");\n    const isWebGL1 = canvas.getContext(\"webgl\");\n    const isWebGL2 = canvas.getContext(\"webgl2\");\n    const isExperimentalWebGL = canvas.getContext(\"experimental-webgl\");\n\n    return is2d ?\n        \"2d\" : isBitmap ?\n            \"bitmaprenderer\" : isWebGL1 ?\n                \"webgl\" : isWebGL2 ?\n                    \"webgl2\" : isExperimentalWebGL ?\n                        \"experimental-webgl\" : null;\n}\n\n\ngdjs._readPixels.getPixelColor = function (canvas, colorIndex, positionX, positionY) {\n    // colorIndex is the index to find in the pixel buffer stored in data constant below.\n    // Values can be 0, 1, 2, 3 for red, green, blue, alpha channel.\n\n    const gl = canvas.getContext(\"webgl2\");\n\n    const data = new Uint8Array(1 * 4); // [255,255,255,255] RGBA\n\n    const invertedPositionY = gl.drawingBufferHeight - positionY;\n    // mandatory because textures in webGL are invert before rendering in the canvas,\n    // otherwise the bottom left corner will be used\n\n    gl.readPixels(positionX, invertedPositionY, 1, 1, gl.RGBA, gl.UNSIGNED_BYTE, data); // Read 1 pixel and store it in data.\n\n    return data[colorIndex];\n\n}\n\n",
```

Comment:
`// getCanvasType, the argument is HTMLCanvasElement`

For this kind of (extremely useful!) information, I recommend that you use the "JSDoc syntax". Advantage: it can be parsed by TypeScript, so you get autocompletion (well, at least in this JS code event block):

```
/** 
 * @param {HTMLCanvasElement} canvas
 */
gdjs._readPixels.getCanvasType = function(canvas) {
...
}
```

> Values can be 0, 1, 2, 3 for red, green, blue, alpha channel

Same, put it a documentation JSDoc block:

```
/**
 * @param {HTMLCanvasElement} canvas
 * @param {number} colorIndex Should be 0, 1, 2 or 3, for red, green, blue or alpha channel.
 * @param {number} positionX The X position of the pixel on the canvas to read
 * @param {number} positionY The Y position of the pixel on the canvas to read
 */
gdjs._readPixels.getPixelColor = function (canvas, colorIndex, positionX, positionY) {
}
```